src/paths/johnson.c

```diff
@@ -115,6 +121,11 @@ igraph_error_t igraph_distances_johnson(const igraph_t *graph,
                      IGRAPH_EINVAL);
     }
 
+    if (mode == IGRAPH_ALL) {
+        IGRAPH_ERROR("Mode IGRAPH_ALL used with negative weight. This will cause a negative loop.",
+                     IGRAPH_EINVAL);
```

Comment:
- Let's make this return `IGRAPH_ENEGLOOP`
 - Let's avoid mentioning C specific things in error messages such as the `IGRAPH_ALL` constant. Instead, I would just say "Negative weight in undirected graph", as mode=ALL implies treating the graph as undirected. Additionally, at the beginning of the function, I would put something like `if (! igraph_is_directed(graph)) mode = IGRAPH_ALL;`

Reply:
I fixed the first two.

For the third: There's already an error for undirected graphs.